Instructor: edit course: make panel responsive #7869

HirdayGupta opened the pull request:
Fixes #7869 

**Outline of solution**

- Overflow was happening in `sm` size screens because of our erroneous usage of bootstrap. Changed the distribution between the dropdown and the button to `5-4` making the total `label-dropdown-button` distribution for `sm` screens `3-5-4`. 
- In `xs` screens the button was sticking because of no explicit handling of xs cases. I added in new classes to handle `xs` cases. The `xs` distribution for `label-button-dropdown` is `12-5-7`. The `xs-12` class for the `label` forces the remaining two elements to the next line. 

**Screenshots of changes:**

 SM Resolution:
<img width="812" alt="screen shot 2017-08-04 at 12 12 02 pm" src="https://user-images.githubusercontent.com/18088721/28957149-4f1edcea-790e-11e7-9a6f-3420694cfad0.png">

XS Resolution:
<img width="395" alt="screen shot 2017-08-04 at 12 12 25 pm" src="https://user-images.githubusercontent.com/18088721/28957155-57e8d128-790e-11e7-93c5-ca5f12cab89f.png">

![ezgif com-video-to-gif-4](https://user-images.githubusercontent.com/18088721/28957750-fa77e2e2-7910-11e7-8fc1-5c034a4a630a.gif)


**Note:** I used the `12-5-7` distribution for `xs` screens because `12-6-6` was resulting in an overflow of the button on the smallest browser window size.

HirdayGupta commented:
Ready for review

LiHaoTan commented:
@HirdayGupta I'm not sure if we should handle `xs` resolutions explicitly as I think <kbd>Auto-Detect Time Zone</kbd> should go onto the next line.

HirdayGupta commented:
@LiHaoTan, the `Auto Detect Time Zone` button being stuck to the bottom of the `select` dropdown was one of the issues brought up in the original issue description. Thus, keeping it on the same line was my solution to that. Are you aware of any way to "un-stick" it and also move it to the next line?

whipermr5 commented:
I agree with @LiHaoTan; it'd look better if the button is on the next line.

> Are you aware of any way to "un-stick" it and also move it to the next line?

Do try it out first.

HirdayGupta commented:
> I agree with @LiHaoTan; it'd look better if the button is on the next line.

I'm not sure I fully follow what the suggestion is here. Are you suggesting to move the button on the next line for all resolutions? Or are you suggesting to only move it to the next line in the xs resolutions?

If it is the first one, it can be done easily. If it is the second one, then I have to keep both the dropdown and the button in the same `boostrap row` so that they display in the same line in all other resolutions, and then for `xs` resolutions, I would give both of them the class `col-xs-12` to move the button to the next line. But this ends up making it like the [`xs` screenshot posted in the issue itself](https://user-images.githubusercontent.com/6811428/28668349-da19eae6-7301-11e7-8d42-6f0c3c24dfea.png), where the button is "stuck" to the bottom part of the dropdown.

whipermr5 commented:
Second one. Find out how to unstick it.

HirdayGupta commented:
@whipermr5 @lihaotan after some research and experimentation with overflows, etc one idea I have is to use @media in CSS to add a class that applies only for xs screen sizes and in that class, to add a padding-top style. Would this be okay? Or did you have something else in mind to unstick it?

HirdayGupta commented:
@whipermr5 Can advise regarding above comment?

HirdayGupta commented:
@whipermr5 Requesting you to please let me know if the method I described in the [above comment](https://github.com/TEAMMATES/teammates/pull/7904#issuecomment-324252881) is suitable. If it isn't I will continue looking for alternatives so that I can get this PR merged quickly.

HirdayGupta commented:
@whipermr5 Opinion please. ^^

whipermr5 commented:
@HirdayGupta Just found this; the hidden div approach here looks cleaner: https://stackoverflow.com/a/19822503

HirdayGupta commented:
@whipermr5 Thanks for that, it was very useful! Never struck me to search for this as "vertical space" 😅. Ready for review!

Summary of changes:
- Added a div that's hidden on all resolutions except `sm`, which contains a blank line.
- Both the time zone dropdown menu and the button are now full width (col-xs-12) on xs resolutions.
- Updated expected  using god mode.

The new xs panel looks like this:
<img width="386" alt="screen shot 2017-08-26 at 1 19 14 pm" src="https://user-images.githubusercontent.com/18088721/29739698-949a290a-8a61-11e7-9876-9dd80a6105c5.png">

### Ready for review

HirdayGupta commented:
@whipermr5 reminder :)

whipermr5 commented:
@HirdayGupta The separation might be a bit too much, especially if in edit mode and the save button is visible:
<img width="384" alt="screen shot 2017-08-28 at 10 39 52 am" src="https://user-images.githubusercontent.com/6811428/29757296-5eb4cd90-8bdd-11e7-866e-df7b5e0c7956.png">

What if you delete the `<br>`? It still gives a 1-pixel spacing, which is not much but does unstick it. What do you think?

HirdayGupta commented:
@whipermr5 you're right. Without the `<br>` looks better especially in edit mode. The <br> was giving it too much spacing and it didn't look like it was part of the time-zone dropdown options. Committed the change.

This is what it looks like in edit mode now:

<img width="371" alt="screen shot 2017-08-28 at 12 16 13 pm" src="https://user-images.githubusercontent.com/18088721/29762351-c0661978-8beb-11e7-8ddf-6a6371451316.png">
